astropy/table/serialize.py

```diff
@@ -110,17 +107,24 @@ def _represent_mixins_as_columns(tbl, exclude_classes=()):
     return a new table.  Exclude any mixin columns in ``exclude_classes``,
     which must be a tuple of classes.
     """
-    if not tbl.has_mixin_columns:
-        return tbl
-
+    # Dict of metadata for serializing each column, keyed by column name.
+    # Gets filled in place by _represent_mixin_as_column().
     mixin_cols = {}
 
+    # List of columns for the output table.  For plain Column objects
+    # this will just be the original column object.
     new_cols = []
 
+    # Go through table columns and represent each column as one or more
+    # plain Column objects (in new_cols) + metadata (in mixin_cols).
     for col in tbl.itercols():
         _represent_mixin_as_column(col, col.info.name, new_cols, mixin_cols,
                                    exclude_classes=exclude_classes)
 
+    # If no metadata was created then just return the original table.
+    if not mixin_cols:
```

Comment:
For code clarity this should probably be named `serialized_cols` now.

Reply:
Yes, if you have the energy, that would be a good change. But perhaps easier to do in a follow-up PR, as this one is large enough already!